references/classification/train_quantization.py

```diff
@@ -46,13 +47,19 @@ def main(args):
 
     print("Creating model", args.model)
     # when training quantized models, we always start from a pre-trained fp32 reference model
-    model = torchvision.models.quantization.__dict__[args.model](weights=args.weights, quantize=args.test_only)
+    model = torchvision.models.quantization.__dict__[args.model](weights=args.weights, quantize=args.test_only, fx_mode=args.fx)
```

Comment:
Though this is the most BC approach for adding support of fx quantization, this API exposes technical details about quantization to the user. Should we be keeping this an internal implementation detail?

Another question I have is related to the use of the various custom nn.Modules (QuantizableMobileNetV2, QuantizableInvertedResidual etc) required by the previous legacy quantization API. My understanding is FX quantization no longer needs this but yet the way we implement fx quantization here still uses those custom modules. Is this necessary? Should we instead create a new set of model builders at `torchvision.prototype.models.quantization` that use the new FX api and don't make use of the custom modules?

Reply:
`FX` is a generic framework which is not specific to quantization.  I think a better name for this flag would be something like `quantization_workflow_type`, which defaults to Eager mode and has FX graph mode quantization as an option.

Reply:
yeah I feel exposing fx as an option to user sounds OK, otherwise, user still need to call a different script to use fx graph mode quantization, this information is exposed anyways

for the custom module for eager mode quantization, just chat with Andrew offline about this, he will change the code to use original floating point model directly in fx mode

Reply:
@datumbox Yes, it's actually not necessary. I think I will modify this to use the original model (e.g. `MobileNetV2`) instead of the one modified for eager mode quantization (e.g. `QuantizableMobileNetV2`) in FX mode. Also, I believe "custom modules" refers to a different, unrelated quantization API that we're not using here.

Reply:
> My understanding is FX quantization no longer needs this but yet the way we implement fx quantization here still uses those custom modules. Is this necessary? Should we instead create a new set of model builders at torchvision.prototype.models.quantization that use the new FX api and don't make use of the custom modules?

maybe we can throw a warning if a user tries to use the eager mode quantizeable modules with the FX graph mode API?

Reply:
@vkuzo Since the latest code doesn't pass the FX flag to the quantized models anymore, I think it'll be a bit complicated to print the warning. Maybe we can do this in a future PR.